### PR TITLE
feat: split new flow

### PR DESCRIPTION
**Pull request by suyashpatil78**
### Description
copilot:summary

copilot:poem

### Walkthrough
copilot:walkthrough

## Clickup
app.clickup.com

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/split-expense/split-expense.page.html [hunk 1]
```diff
@@ -13,7 +13,7 @@
         </div>
         <ion-buttons>
           <ion-button
-            (click)="save()"
+            (click)="saveV2()"
```
Comment (suyashpatil78) [inline src/app/fyle/split-expense/split-expense.page.html:16]: Will rename this to again save() after all changes are done

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/core/services/split-expense.service.ts [hunk 2]
```diff
@@ -319,4 +364,57 @@ export class SplitExpenseService {
...
+    const splitPolicyChecks$ = this.handleSplitPolicyCheck(splitEtxns, fileObjs, originalTxn, reportId);
+    const splitMissingFieldsCheck$ = this.handleSplitMissingFieldsCheck(splitEtxns, fileObjs, originalTxn, reportId);
+
+    return forkJoin({ policyViolations: splitPolicyChecks$, missingFields: splitMissingFieldsCheck$ });
+  }
+
+  getFileIdsFromObjects(fileObjs: FileObject[]): string[] {
+    const fileIds = [];
+
+    if (fileObjs && fileObjs.length > 0) {
```
Comment (Dimple16) [inline src/app/core/services/split-expense.service.ts:412]: `fileObjs?.length`

**Review by Dimple16 — APPROVED** [review]
minor comments

File: src/app/core/services/platform/v1/spender/expenses.service.ts [hunk 3]
```diff
@@ -113,6 +118,106 @@ export class ExpensesService {
     });
   }
 
+  formatSplitResponse(response) {
+    const formattedResponse = {};
+
+    for (let index = 0; index < response?.data?.length; index++) {
```
[the remark this answers is not in the document]
Comment (suyashpatil78) [reply src/app/core/services/platform/v1/spender/expenses.service.ts]: I have removed this method as it is not used anymore

File: src/app/core/services/split-expense.service.ts (hunk 2 above)
Comment (suyashpatil78) [reply src/app/core/services/split-expense.service.ts]: done

**Pull request merged**